app.js

```diff
@@ -24,14 +27,30 @@ app.get("/api/articles/:article_id", getArticleByID);
 
 app.get("/api/articles", getArticles);
 
+app.post("/api/articles/:article_id/comments", postComment)
+
 app.all('/*', (req,res) => {
     res.status(404).send({msg: "path not found"})
 })
 
 app.use((err,req,res,next) => {
-    if(err.code === '22P02') {
+    //console.log(err)
+    //console.log(err.code)
+    if(err.status && err.msg) {
+        res.status(err.status).send({msg: err.msg})
+    }
+    if(err.length === 240) {
         res.status(400).send({msg: 'Bad Request'})
     }
+    if(err.constraint === 'comments_article_id_fkey') {
+        res.status(404).send({msg: 'not found'})
+    }
+    if(err.code === '22P02'|| err.code === '23503') {
+        res.status(400).send({msg: 'Bad Request'})
+    }
+    if(err.code === '23502') {
```

Comment:
Here's your issue, `23502` and `23503` need to be swapped around:
- `23502` - violation of a 'not null' constraint which is _400_
- `23503` - foreign key violation which is _404_